test/upgradeableProxy/Dispatcher.upgrade.t.sol

```diff
@@ -36,43 +45,67 @@ abstract contract UpgradeTestUtils {
 contract ChannelHandShakeUpgradeUtil is ChannelHandshakeUtils {
     uint32 nextSequenceSendSlot = 153;
     uint32 sendPacketCommitmentSlot = 156;
+    uint32 ackPacketCommitmentSlot = 158;
     uint32 nextSequenceAckSlot = 155;
     uint32 nextSequenceRecvSlot = 154;
     IbcPacket[3] packets;
     string payload = "msgPayload";
     bytes packet = abi.encodePacked(payload);
     uint64 timeoutTimestamp = 1000;
 
-    function doChannelHandshake() public {
+    // Conduct 4-step channel handshake between local and remote end
+    // Have to pass in receivingDapp because it's hard to parse out the addresss from the portId
+    function doChannelHandshake(LocalEnd memory le, ChannelEnd memory re, IbcChannelReceiver receivingDapp) public {
         // same setup as that run in test_connectChannel_ok
         ChannelHandshakeSetting[8] memory settings = createSettings2(true);
 
+        LocalEnd memory remote =
+            LocalEnd(receivingDapp, re.portId, re.channelId, le.connectionHops, le.versionCall, le.versionExpected);
+
+        ChannelEnd memory local = ChannelEnd(le.portId, le.channelId, le.versionCall);
         string[2] memory versions = ["1.0", "2.0"];
         for (uint256 i = 0; i < settings.length; i++) {
             for (uint256 j = 0; j < versions.length; j++) {
-                LocalEnd memory le = _local;
-                ChannelEnd memory re = _remote;
+                // mockHandShakeProofs(le, re, settings[i], versions[j]);
                 le.versionCall = versions[j];
                 le.versionExpected = versions[j];
                 re.version = versions[j];
+                remote.versionCall = versions[j];
+                remote.versionExpected = versions[j];
+                local.version = versions[j];
                 channelOpenInit(le, re, settings[i], true);
-                channelOpenTry(le, re, settings[i], true);
+                channelOpenTry(remote, local, settings[i], true);
                 channelOpenAck(le, re, settings[i], true);
-                channelOpenConfirm(le, re, settings[i], true);
+                channelOpenConfirm(remote, local, settings[i], true);
```

Comment:
why do we need `le`,`re`, `local` and `remote`? wouldn't `le`,`re` be enough?

Reply:
the 4 channel helpers here take a LocalEnd as the first type and a ChannelEnd as the second type. Because we want to flip-flop passing in the local and remote channel ends to these 4 handshake helper methods, we need the `local` and `remote` to store the other analagous types.

- `le` is the local end as a `LocalEnd` type
- `local` is the local end as a `ChannelEnd` type
- `re` is the remote end as a `ChannelEnd` type
- `remote` is the remote end as a `LocalEnd` type

Reply:
consider renaming them to 
- `le` -> `localEnd`
- `local` -> `localChannelEnd`
- `re` -> `remoteChannelEnd`
- `remote` -> `remoteEnd`

Reply:
should be renamed now